sources/core/objects/Floor.cpp

```diff
@@ -28,7 +28,6 @@ void Floor::AfterWorldCreation()
 {
     ITurf::AfterWorldCreation();
     SetOpen(open_);
-    SetState(floor_type_);
```

Comment:
Because an object creation from mapgen works the next way:
1. It created and its constructor called (default `floor_type_` is set here)
2. Variables for this object from map are set (here it is possible that `floor_type_` is set to some new cool color)
3. `AfterWorldCreation` is called.

We cannot set sprite states now in the map editor. That means even if `floor_type_` is set in the map editor the floor object still will have a default state. The state will be changed only when the floor will be opened and closed again. 
So in order to bypass that there is `SetState(floor_type_)` call in the `AfterWorldCreation()` functions. It is hack, but for a while it is works that way.